gobblin-runtime/src/main/java/org/apache/gobblin/runtime/Task.java

```diff
@@ -948,4 +953,12 @@ public boolean isSpeculativeExecutionSafe() {
     }
     return true;
   }
+
+  public void setTaskFuture(Future<?> taskFuture) {
+    this.taskFuture = taskFuture;
+  }
+
+  public Future<?> getTaskFuture() {
```

Comment:
Adding this method  and making `completeShutdown` public is not needed if we add a `cancel()` method to `Task` like: 
```java
  public boolean cancel() {
    if (this.taskFuture != null) {
      if (this.taskFuture.cancel(true)) {
        this.completeShutdown();
      }
    }
    return false;
  }
```

Reply:
done